src/main/java/orlov/programming/springcoregym/facade/impl/GYMFacadeImpl.java

```diff
@@ -1,70 +1,248 @@
 package orlov.programming.springcoregym.facade.impl;
 
+import lombok.extern.log4j.Log4j2;
 import org.springframework.stereotype.Component;
+import orlov.programming.springcoregym.bootstrap.TrainingTypeLoader;
 import orlov.programming.springcoregym.facade.GYMFacade;
 import orlov.programming.springcoregym.model.training.Training;
+import orlov.programming.springcoregym.model.training.TrainingType;
 import orlov.programming.springcoregym.model.user.Trainee;
 import orlov.programming.springcoregym.model.user.Trainer;
+import orlov.programming.springcoregym.model.user.User;
 import orlov.programming.springcoregym.service.training.TrainingService;
 import orlov.programming.springcoregym.service.user.trainee.TraineeService;
 import orlov.programming.springcoregym.service.user.trainer.TrainerService;
 
+import java.time.LocalDate;
+import java.util.List;
+
+@Log4j2
 @Component
 public class GYMFacadeImpl implements GYMFacade {
 
     private final TraineeService traineeService;
     private final TrainerService trainerService;
     private final TrainingService trainingService;
+    private final TrainingTypeLoader trainingTypeLoader;
 
+    private User loggedUser;
 
-    public GYMFacadeImpl(TraineeService traineeService, TrainerService trainerService, TrainingService trainingService) {
+    public GYMFacadeImpl(TraineeService traineeService, TrainerService trainerService, TrainingService trainingService,
+                         TrainingTypeLoader trainingTypeLoader) {
         this.traineeService = traineeService;
         this.trainerService = trainerService;
         this.trainingService = trainingService;
+        this.trainingTypeLoader = trainingTypeLoader;
+        trainingTypeLoader.loadDefaultTrainingTypes();
     }
 
     @Override
     public Trainee createTrainee(Trainee trainee) {
-        return traineeService.create(trainee);
+        try {
+            return traineeService.create(trainee);
+        } catch (Exception e) {
+            log.error(e);
+            return null;
```

Comment:
It is bad practice to return null
https://medium.com/javarevisited/just-dont-return-null-dcdf5d77128f